modulegen/internal/workfile/main.go

```diff
@@ -0,0 +1,65 @@
+package workfile
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"golang.org/x/mod/modfile"
+
+	"github.com/testcontainers/testcontainers-go/modulegen/internal/context"
+	"github.com/testcontainers/testcontainers-go/modulegen/internal/module"
+)
+
+type Generator struct{}
+
+// Generate updates github ci workflow
+func (g Generator) Generate(ctx context.Context) error {
+	examples, modules, err := module.ListExamplesAndModules(ctx)
+	if err != nil {
+		return err
+	}
+
+	rootDir := ctx.RootDir
+
+	rootGoWorkFilePath := filepath.Join(rootDir, "go.work")
+
+	rootGoWork, err := Read(rootGoWorkFilePath)
+	if err != nil {
+		return fmt.Errorf("read go.mod file: %w", err)
+	}
+
+	err = newWorkFile(rootGoWork, examples, modules)
+	if err != nil {
+		return fmt.Errorf("create go.mod file: %w", err)
+	}
+
+	return Write(rootGoWorkFilePath, rootGoWork)
+}
+
+func newWorkFile(goWork *modfile.WorkFile, examples []string, modules []string) error {
```

Comment:
question: is the intent to exclude other modules like test, and wait? If not using a find for `go.mod` might be simpler.

Reply:
but wait is a package, not a module, right?